src/promptflow/promptflow/azure/operations/_run_operations.py

```diff
@@ -1037,6 +1044,16 @@ def download(
             # Reference: https://stackoverflow.com/questions/45600579/asyncio-event-loop-is-closed-when-getting-loop
             # On Windows seems to be a problem with EventLoopPolicy, use this snippet to work around it
             asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
-        asyncio.run(run_downloader.download())
-        print(f"Successfully downloaded run {run!r} to {run_folder.resolve().as_posix()!r}.")
-        return run_folder.resolve().as_posix()
+
+        # when running in notebook, there is already a running event loop, we need to handle this case
+        try:
```

Comment:
I thought there is a util function for this, may leverage that

Reply:
good point, updated.